src/main/kotlin/com/terraformation/backend/tracking/db/T0Store.kt

```diff
@@ -540,22 +457,168 @@ class T0Store(
     }
   }
 
-  private fun plotHasCompletedObservations(
-      monitoringPlotIdField: Field<MonitoringPlotId?>,
-      isPermanent: Field<Boolean?>,
-  ): Condition =
-      DSL.exists(
-          DSL.selectOne()
-              .from(OBSERVATION_PLOTS)
-              .where(
-                  OBSERVATION_PLOTS.MONITORING_PLOT_ID.eq(monitoringPlotIdField)
-                      .and(OBSERVATION_PLOTS.IS_PERMANENT.eq(isPermanent))
-                      .and(OBSERVATION_PLOTS.COMPLETED_TIME.isNotNull)
+  private fun getAllPermPlotSpeciesT0Data(plantingSiteId: PlantingSiteId): Pair<Boolean, Boolean> {
+
+    data class T0SitePermPlotData(
+        val monitoringPlotId: MonitoringPlotId,
+        val survivalRateIncludesTempPlots: Boolean,
+        val speciesId: SpeciesId?,
+        val plotDensity: BigDecimal? = null,
+    )
+
+    with(MONITORING_PLOTS) {
+      val permanentPlotSpecies =
+          DSL.select(
+                  ID.`as`("plot_id"),
+                  PLANTING_SUBZONE_POPULATIONS.SPECIES_ID.`as`("species_id"),
               )
-              .orderBy(
-                  OBSERVATION_PLOTS.COMPLETED_TIME.desc(),
-                  OBSERVATION_PLOTS.OBSERVATION_ID.desc(),
+              .from(this)
+              .join(PLANTING_SUBZONE_POPULATIONS)
+              .on(PLANTING_SUBZONE_ID.eq(PLANTING_SUBZONE_POPULATIONS.PLANTING_SUBZONE_ID))
+              .where(PLANTING_SITE_ID.eq(plantingSiteId))
+              .unionAll(
+                  DSL.select(
+                          ID.`as`("plot_id"),
+                          PLOT_T0_DENSITIES.SPECIES_ID.`as`("species_id"),
+                      )
+                      .from(this)
+                      .join(PLOT_T0_DENSITIES)
+                      .on(ID.eq(PLOT_T0_DENSITIES.MONITORING_PLOT_ID))
+                      .where(PLANTING_SITE_ID.eq(plantingSiteId))
               )
-              .limit(1)
-      )
+              .asTable("permanent_plot_species")
+
+      val permPlotIdField = permanentPlotSpecies.field("plot_id", ID.dataType)
+      val permSpeciesIdField = permanentPlotSpecies.field("species_id", SpeciesId::class.java)!!
+
+      val permPlotData =
+          dslContext
+              .select(
+                  ID,
+                  plantingSites.SURVIVAL_RATE_INCLUDES_TEMP_PLOTS,
+                  permSpeciesIdField,
+                  PLOT_T0_DENSITIES.PLOT_DENSITY,
+              )
+              .from(this)
```

Comment:
I had to scroll up to see which table `this` was referring to, and the `plantingSites` implicit join made me think there was a local variable by that name somewhere. I think this would be more readable if each query had its own `with`.